onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -3507,10 +3507,39 @@ def aten_linear_backward(
     raise NotImplementedError()
 
 
-def aten_linspace(start: float, end: float, steps: int) -> TensorType:
+@torch_op("aten::linspace", trace_only=True)
+def aten_linspace(start: TRealUnlessFloat16OrInt8, end: TRealUnlessFloat16OrInt8, steps: int, dtype: int = -1) -> TensorType:
```

Comment:
I am now pondering if we should split the dtype variation out to avoid traceonly and leverage the dispatcher

Reply:
Yes, this should be a general solution for other ops as well: let dispatcher to handle this logic of choosing correct overload.

Reply:
Sounds good. We can do a follow up pass to clean this up. @titaiwangms fyi - what are your thoughts?

Reply:
Not sure I follow:

The scenario is:
1. Let dispatcher do the work: Two scripted overloads that one is with dtype, and one is not.
2. Use trace_only: Only one overload, so there is nothing to dispatch.

In this case, dispatcher doesn't have to do anything.

Reply:
> Not sure I follow:
> 
> The scenario is:
> 
> 1. Let dispatcher do the work: Two scripted overloads that one is with dtype, and one is not.
> 2. Use trace_only: Only one overload, so there is nothing to dispatch.
> 
> In this case, dispatcher doesn't have to do anything.

The scenario should be:
1, torch_lib provides 2 scripted overloads, one is with dtype and the other is not., No trace_only here.
2, Dispatcher needs to decide which one to call for export, depending on what dispatcher got from FX graph.

Reply:
SG! I remember we already have those ops! The dispatcher can do that.